src/js/spatial-navigation.js

```diff
@@ -0,0 +1,496 @@
+  stop() {
+    this.player.el().removeEventListener('keydown', this.onKeyDown);
+    this.isListening = false;
+  }
+
+  /**
+   * Responds to keydown events for spatial navigation and media control.
+   *
+   * Determines if spatial navigation or media control is active and handles key inputs accordingly.
+   *
+   * @param {KeyboardEvent} e - The keydown event to be handled.
+   */
+  onKeyDown(e) {
+    const key = this.KEY_CODE[e.keyCode];
+
+    if (!this.isPaused && ['up', 'right', 'down', 'left'].includes(key)) {
+      this.move(key);
+      e.preventDefault();
+    } else if (['play', 'pause', 'ff', 'rw'].includes(key)) {
+      this.performMediaAction(key);
+      e.preventDefault();
+    }
+  }
+
+  /**
+   * Performs media control actions based on the given key input.
+   *
+   * Controls the playback and seeking functionalities of the media player.
+   *
+   * @param {string} key - The key representing the media action to be performed.
+   *   Accepted keys: 'play', 'pause', 'ff' (fast-forward), 'rw' (rewind).
+   */
+  performMediaAction(key) {
+    if (this.player) {
+      switch (key) {
+      case 'play':
+        if (this.player.paused()) {
+          this.player.play();
+        }
+        break;
+      case 'pause':
+        if (!this.player.paused()) {
+          this.player.pause();
+        }
+        break;
+      case 'ff':
+        this.userSeek_(this.player.currentTime() + this.STEP_SECONDS);
+        break;
+      case 'rw':
+        this.userSeek_(this.player.currentTime() - this.STEP_SECONDS);
+        break;
+      default:
+        break;
+      }
+    }
+  }
+
+  /**
+   * Prevent liveThreshold from causing seeks to seem like they
+   * are not happening from a user perspective.
+   *
+   * @param {number} ct
+   *        current time to seek to
+   */
+  userSeek_(ct) {
+    if (this.player.liveTracker && this.player.liveTracker.isLive()) {
+      this.player.liveTracker.nextSeekedFromUser();
+    }
+
+    this.player.currentTime(ct);
+  }
+
+  /**
+   * Pauses the spatial navigation functionality.
+   * This method sets a flag that can be used to temporarily disable the navigation logic.
+   */
+  pause() {
+    this.isPaused = true;
+  }
+
+  /**
+   * Resumes the spatial navigation functionality if it has been paused.
+   * This method resets the pause flag, re-enabling the navigation logic.
+   */
+  resume() {
+    this.isPaused = false;
+  }
+
+  /**
+   * Handles Player Blur.
+   *
+   */
+  handlePlayerBlur(component) {
+    if (component.name() === 'CloseButton') {
+      this.refocusComponent();
+    } else {
+      this.pause();
+
+      if (component && component.el_) {
+        this.lastFocusedComponent = component;
+      }
+    }
+  }
+
+  /**
+   * Handles Player Focus.
+   *
+   */
+  handlePlayerFocus() {
+    this.resume();
+  }
+
+  /**
+   * Gets a set of focusable components.
+   *
+   * @return {Array}
+   *         Returns an array of focusable components.
+   */
+  getComponents() {
```

Comment:
I think the naming and documentation of this function is a little confusing when this intention is actually to update `this.focusableComponents`.. maybe it could be renamed to `updateFocusableComponents`..

Then, getComponents could just return `this.this.focusableComponents`

Reply:
Made a commit to rename 'focusableComponents' to 'updateFocusableComponents', regarding "getComponents could just return this.this.focusableComponents" I think is a good idea & we can implement it when we required to just return the focusable components.